presentation-compiler/src/main/dotty/tools/pc/completions/Completions.scala

```diff
@@ -228,33 +226,47 @@ class Completions(
   def completionsWithSuffix(
       denot: SingleDenotation,
       label: String,
-      toCompletionValue: (String, SingleDenotation, CompletionSuffix) => CompletionValue
+      toCompletionValue: (String, SingleDenotation, CompletionAffix) => CompletionValue
   ): List[CompletionValue] =
     val sym = denot.symbol
-    // find the apply completion that would need a snippet
+    val hasNonSyntheticConstructor = sym.name.isTypeName && sym.isClass
+      && !sym.is(ModuleClass) && !sym.is(Trait) && !sym.is(Abstract) && !sym.is(Flags.JavaDefined)
+
     val methodDenots: List[SingleDenotation] =
-      if shouldAddSnippet && completionMode.is(Mode.Term) &&
-        (sym.is(Flags.Module) || sym.isField || sym.isClass && !sym.is(Flags.Trait)) && !sym.is(Flags.JavaDefined)
-      then
-        val info =
-          /* Companion will be added even for normal classes now,
-           * but it will not show up from classpath. We can suggest
-           * constructors based on those synthetic applies.
-           */
-          if sym.isClass && sym.companionModule.exists then sym.companionModule.info
-          else denot.info
-        val applyDenots = info.member(nme.apply).allSymbols.map(_.asSeenFrom(info).asSingleDenotation)
-        denot :: applyDenots
+      if shouldAddSnippet && isNew && hasNonSyntheticConstructor then
+        sym.info.member(nme.CONSTRUCTOR).allSymbols.map(_.asSingleDenotation)
+          .filter(_.symbol.isAccessibleFrom(denot.info))
```

Comment:
I'm not sure that you're filtering using the correct scope here (also line `254`).

Reply:
I've added a test that checks if private members are correctly filtered, and they are failing without this line

In my opinion we're filtering with correct scope, as `denot.info` is the member in a specific call site.